src/ride/track.c

```diff
@@ -370,7 +370,7 @@ void track_list_populate(ride_list_item item, uint8* track_list_cache){
 		}
 
 		// If cur_track_entry_index is greater than max number of tracks
-		if (cur_track_entry_index >= 1000){
+		if (cur_track_entry_index >= 254){
```

Comment:
I think so... the peep direction looks like it could be a severe bug so that's a good find.